### PR TITLE
Try to explain non-struct types in the chessboard exercise (#1602).

### PR DESCRIPTION
**Pull request by bobtfish**
I've had a crack at this. I also added methods as a prerequisite to this exercise as it uses them so
they should probably be covered first.

### CODE REVIEW

File: exercises/concept/chessboard/.docs/introduction.md [hunk 1]
```diff
@@ -92,3 +91,33 @@ for range xi {
 // count value:
 // 3
 ```
+
+You've previously seen defining struct types, but it's also possible to define non
```
Comment (jmrunkle) [inline exercises/concept/chessboard/.docs/introduction.md:95]: Prefer to put line breaks at sentence breaks. In both the actual display and the review display for this the lines wrap and doing it like this just increases the blast radius for lines changed when we do reviews. For example, removing or changing one or two words ends up meaning you have to reformat the whole paragraph.

File: exercises/concept/chessboard/.docs/introduction.md [hunk 2]
```diff
@@ -92,3 +91,33 @@ for range xi {
...
+struct types which you can use as an alias for a built in type declaration, and
+you can define reviever functions on them to extend them in the same way as struct types.
```
Comment (jmrunkle) [inline exercises/concept/chessboard/.docs/introduction.md:97]: typo: reviever

File: exercises/concept/chessboard/.docs/introduction.md [hunk 3]
```diff
@@ -92,3 +91,33 @@ for range xi {
 // count value:
 // 3
 ```
+
```
Comment (jmrunkle) [inline exercises/concept/chessboard/.docs/introduction.md:94]: Make a new header:

```
## Non-struct types
```

File: exercises/concept/chessboard/.docs/introduction.md (hunk 1 above)
Comment (jmrunkle) [reply exercises/concept/chessboard/.docs/introduction.md:95]: nit: "non" is not a word on its own, so hyphenate: non-struct

**Review by jmrunkle — CHANGES_REQUESTED** [review]
Thanks for working on this. Just asking for a bit of cleanup before we merge this.

**Comment by jmrunkle** [comment]
RE: the configlet lint failure, it should be fixed on the main branch. If you sync your fork and rebase you should be good to go.

**Comment by bobtfish** [comment]
> RE: the configlet lint failure, it should be fixed on the main branch. If you sync your fork and rebase you should be good to go.

Thanks for the feedback <3

**Review by jmrunkle — APPROVED** [review]
Thanks!

**Pull request merged**